README.md

```diff
@@ -240,30 +240,38 @@ The workaround is simply to do one of two things:
 
 #### Service and controller injections
 
-Ember does service and controller lookups with the `inject` helpers at runtime, using the name of the service or controller being injected up as the default value—a clever bit of metaprogramming that makes for a nice developer experience. TypeScript cannot do this, because the name of the service or controller to inject isn't available at compile time in the same way. This means that if you do things the normal Ember way, you will have to specify the type of your service or controller explicitly everywhere you use it.
+In Ember 1.x and Ember 2.x we did service and controller lookups in `EmberObjects` with the `inject` helpers at runtime, using the name of the service or controller being injected up as the default value. TypeScript cannot do this, because the name of the service or controller to inject isn't available at compile time in the same way. This means that we need to do things differently in Ember 3.x with ES6 classes.
+
+We also want the injection to be per-class, not per-instance (slower), unlike the `inject` helpers in ES6. Lukily the hard work has been done for us. [Ember-decorators](https://github.com/ember-decorators/ember-decorators) to the rescue! This declarative addon makes injections trivially easy.
```

Comment:
I think if we're going to put this type of technical detail in it should go somewhere else. This document should be a "how to make it work" document - it's where new-to-TS developers will be looking and we need it to look as easy as it is. The hard work should be done by the generators and the addons.

Reply:
Totally agree, but at this point, the confusion is already a problem since some developers are already down the wrong path. Perhaps a well-reviewed blog article would be better suited for the technical details. At minimum, I would ask that you be 100% clear that this is not a special thing for TS or ES6 ember users -- it's in alignment with how things have always been.

Reply:
I'll note in connection with this that I fully intend to split apart the docs at a later point – see #170 for that plan – but *until* then, I'd like to leave what's in the README in the README. I'll do another review of the changes later this afternoon.